charts/common/templates/_helpers.tpl

```diff
@@ -98,14 +98,38 @@ Either gcr.io/cloudsql-docker/gce-proxy:1.* or with gcr.io/cloud-sql-connectors/
 
 {{/*
 Create common.instanceConnectionName depending on .Values.cloudsqlProxy.instanceConnectionName
-or .Values.cloudsqlProxy.migrationTemplate.instanceConnectionName
+else if .Values.cloudsqlProxy.migrationTemplate.instanceConnectionName
+else if try to extract it from the .Values.cloudsqlProxy.command list
+else if try to extract it from the .Values.cloudsqlProxy.args list
+else fail.
+This variable will be used in deployment.yaml and workflowtemplate.yaml it's a safer way to get the instanceConnectionName.
+Once all apps are using cloud sql proxy v2 this can be simplified.
 */}}
 {{- define "common.instanceConnectionName" -}}
-{{- if .Values.cloudsqlProxy.instanceConnectionName }}
-{{- printf .Values.cloudsqlProxy.instanceConnectionName }}
-{{- else if .Values.cloudsqlProxy.migrationTemplate.instanceConnectionName }}
-{{- printf .Values.cloudsqlProxy.migrationTemplate.instanceConnectionName }}
-{{- else }}
-{{- fail "instanceConnectionName is not set" }}
-{{- end }}
-{{- end }}
+  {{- if .Values.cloudsqlProxy.instanceConnectionName -}}
+    {{- .Values.cloudsqlProxy.instanceConnectionName  -}}
+  {{- else if .Values.cloudsqlProxy.migrationTemplate.instanceConnectionName -}}
+    {{- .Values.cloudsqlProxy.migrationTemplate.instanceConnectionName  -}}
+  {{- else if .Values.cloudsqlProxy.command -}}
+    {{- $commandList := .Values.cloudsqlProxy.command -}}
+    {{- $instanceArg := "" -}}
+    {{- range $cmd := $commandList -}}
+      {{- if hasPrefix "-instances=" $cmd -}}
+        {{- $instanceArg = $cmd -}}
+      {{- end -}}
+    {{- end -}}
+    {{- trimPrefix "-instances=" $instanceArg }}
+  {{- else if .Values.cloudsqlProxy.args -}}
+    {{- $argsList := .Values.cloudsqlProxy.args -}}
```

Comment:
I'm confused why processing `args` and `command` is different?

Reply:
Because we parse each of them differently to get the `internal-1-4825:us-central1:staging-2` part
```
    args:
      - "--private-ip"
      - "internal-1-4825:us-central1:staging-2?unix-socket-path=/cloudsql/internal-1-4825:us-central1:staging-2"```
and
```
    command:
      - /cloud_sql_proxy
      - "--dir=/cloudsql"
      - "-instances=internal-1-4825:us-central1:staging-2"
      - "-verbose=false"```

Reply:
Oh its a v1 vs v2 syntax diff.